zerver/views/user_groups.py

```diff
@@ -183,22 +182,6 @@ def edit_user_group(
     return json_success(request)
 
 
-@require_member_or_admin
-@typed_endpoint
-def delete_user_group(
-    request: HttpRequest,
-    user_profile: UserProfile,
-    *,
-    user_group_id: PathOnly[Json[int]],
-) -> HttpResponse:
-    # For deletion, the user group's recursive subgroups and the user group itself are locked.
-    with lock_subgroups_with_respect_to_supergroup(
-        [user_group_id], user_group_id, acting_user=user_profile
-    ) as context:
-        check_delete_user_group(context.supergroup, acting_user=user_profile)
-    return json_success(request)
```

Comment:
Does `deactivate_user_group` deserve this same locking strategy?

Reply:
I am not sure. The group which is being deactivated is locked, and I guess that should be enough, to prevent any other concurrent requests breaking the DAG constraint.

Reply:
OK